src/parcsr_mv/par_csr_fffc_device.c

```diff
@@ -264,12 +264,21 @@ hypre_ParCSRMatrixGenerateFFFCDevice_core( hypre_ParCSRMatrix  *A,
    nF_local = n_local - nC_local;
    nF_global = hypre_ParCSRMatrixGlobalNumRows(A) - nC_global;
 
-   CF_marker  = hypre_TAlloc(HYPRE_Int,    n_local,         HYPRE_MEMORY_DEVICE);
    map2FC     = hypre_TAlloc(HYPRE_Int,    n_local,         HYPRE_MEMORY_DEVICE);
    itmp       = hypre_TAlloc(HYPRE_Int,    n_local,         HYPRE_MEMORY_DEVICE);
    recv_buf   = hypre_TAlloc(HYPRE_BigInt, num_cols_A_offd, HYPRE_MEMORY_DEVICE);
 
-   hypre_TMemcpy(CF_marker, CF_marker_host, HYPRE_Int, n_local, HYPRE_MEMORY_DEVICE, HYPRE_MEMORY_HOST);
+   hypre_MemoryLocation cf_memory_location;
+   hypre_GetPointerLocation(CF_marker_host, &cf_memory_location);
```

Comment:
It's OK to leave it as is for now, since @waynemitchell is going to put `CF_marker` in `hypre_Intarray` so we will know the memory location. `hypre_GetPointerLocation` should not be used unless inside `#if HYPRE_DEBUG`. Current CUDA generates errors when getting attributes of host pointers and we kind of rely on that to tell it is a host pointer.

Reply:
@liruipeng @ulrikeyang @rfalgout What would be your preference here? Shall I leave this as it is for this PR and fix it in PR #434? Or should I try to merge the hypre_IntArray stuff into this PR? I guess I would lean towards keeping them separate for now (i.e. merge this PR as is and leave the longer-term fix of using hypre_IntArray for later).

Reply:
I am fine with keeping them separate for now.